crates/breez-sdk/core/src/utils/send_payment_validation.rs

```diff
@@ -0,0 +1,605 @@
+use crate::{InputType, SparkInvoiceDetails, error::SdkError, models::PrepareSendPaymentRequest};
+use web_time::{Duration, SystemTime, UNIX_EPOCH};
+
+/// Validates a send payment request against the parsed input type.
+pub(crate) fn validate_prepare_send_payment_request(
+    input_type: &InputType,
+    request: &PrepareSendPaymentRequest,
+    identity_public_key: &str,
+) -> Result<(), SdkError> {
+    match input_type {
+        InputType::SparkInvoice(spark_invoice_details) => {
+            validate_spark_invoice_request(spark_invoice_details, request, identity_public_key)
+        }
+        InputType::SparkAddress(_) => validate_spark_address_request(request),
+        InputType::Bolt11Invoice(_) => validate_bolt11_invoice_request(request),
+        InputType::BitcoinAddress(_) => validate_bitcoin_address_request(request),
+        _ => Err(SdkError::InvalidInput(
+            "Unsupported payment method".to_string(),
+        )),
+    }
+}
+
+/// Validates a spark invoice request against the provided request parameters.
+fn validate_spark_invoice_request(
+    spark_invoice_details: &SparkInvoiceDetails,
+    request: &PrepareSendPaymentRequest,
+    identity_public_key: &str,
+) -> Result<(), SdkError> {
+    // Validate token identifier
+    if let Some(token_identifier) = &spark_invoice_details.token_identifier {
+        if let Some(requested_token_identifier) = &request.token_identifier {
+            if requested_token_identifier != token_identifier {
+                return Err(SdkError::InvalidInput(
+                    "Requested token identifier does not match invoice token identifier"
+                        .to_string(),
+                ));
+            }
+        } else {
+            return Err(SdkError::InvalidInput(
+                "Token identifier is required for tokens invoice".to_string(),
+            ));
```

Comment:
I can understand validating if the request token_identifier matches the invoice if set, but is there a reason to force the request token_identifier to be set in this case? It's already in the invoice